server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/MailboxQuerySerializer.scala

```diff
@@ -40,7 +40,12 @@ object MailboxQuerySerializer {
       .getOrElse(JsError(s"$value is not a valid role"))
     case _ => JsError("Expecting a JsString to be representing a role")
   }
-  private implicit val filterReads: Reads[MailboxFilter] = Json.reads[MailboxFilter]
+
+  private implicit val filterReads: Reads[MailboxFilter] =  {
+    case JsObject(underlying) if underlying.keySet.diff(MailboxFilter.SUPPORTED).nonEmpty => JsError(s"Unsupported filter")
```

Comment:
Can we say which filters were unsupported?

`JsError(s"Unsupported filter ${underlying.keySet.diff(MailboxFilter.SUPPORTED).toSting}")`